lib/utils.js

```diff
@@ -1,24 +1,27 @@
-var version = require('../package.json').version
-var phin = require('phin')
+const version = require('../package.json').version
+const phin = require('phin').unpromisified
 
-var headers = {
+const headers = {
   'User-Agent': 'node-yggdrasil/' + version,
   'Content-Type': 'application/json'
 }
 
-var utils = {}
+const utils = {}
 
 utils.phin = phin
 
 /**
  * Generic POST request
  */
-utils.call = function (host, path, data, cb) {
+utils.call = function (host, path, data, cb, agent) {
   phin({
     method: 'POST',
-    url: host + '/' + path,
+    url: `${host}/${path}`,
     data: data,
-    headers: headers
+    headers: headers,
+    core: {
+      agent: agent
```

Comment:
is an undefined agent well defined ?

Reply:
Yep, that's the default behaviour:
https://nodejs.org/api/http.html#http_http_request_url_options_callback

Reply:
I'd also say that you can write `headers: headers,` as just `headers,`